### PR TITLE
Add productivity rate and the post processor.

### PR DESCRIPTION
**Pull request by ismailsunni**
### What does it fix?
* Ticket: #3626
* Funded by: DFAT
* Description: 
   - Adding productivity rate and the post processor.

landcover_productivity

Hi @fredychandra may be you can try this.

Just a note, the analysis is done, but since I am no expert in the reporting, I will leave it to @myarjunar 

Some wording may be need to be polished.

I have add the currency as a InaSAFE setting. I think it's more suitable to make it as global setting rather than per layer's metadata. Most use cases people do not change the currency for every layer.

Currently, I only added 5 currencies. We can add more later if needed.

currency

### CODE REVIEW

File: safe/definitions/post_processors/__init__.py [hunk 1]
```diff
@@ -43,5 +43,6 @@
      age_vulnerability_postprocessors +
      disabled_vulnerability_postprocessors +
      gender_vulnerability_postprocessors) + [
-    post_processor_additional_rice
+    post_processor_additional_rice,
```
Comment (ismailsunni) [inline safe/definitions/post_processors/__init__.py]: __[flake8]__
* [F405] 'post_processor_additional_rice' may be undefined, or defined from star imports: minimum_needs_post_processors, population_post_processors, post_processor_functions, post_processor_inputs, post_processors

File: safe/definitions/post_processors/__init__.py [hunk 2]
```diff
@@ -43,5 +43,6 @@
...
+    post_processor_productivity
```
Comment (ismailsunni) [inline safe/definitions/post_processors/__init__.py]: __[flake8]__
* [F405] 'post_processor_productivity' may be undefined, or defined from star imports: minimum_needs_post_processors, population_post_processors, post_processor_functions, post_processor_inputs, post_processors

File: safe/definitions/post_processors/__init__.py [hunk 3]
```diff
@@ -1,9 +1,11 @@
 # coding=utf-8
-from population_post_processors import *
-from post_processors import *
+from safe.definitions.post_processors.productivity_post_processors import \
+    productivity_post_processors
 from minimum_needs_post_processors import *
-from post_processor_inputs import *
+from population_post_processors import *
```
Comment (ismailsunni) [inline safe/definitions/post_processors/__init__.py]: __[flake8]__
* [F403] 'from population_post_processors import *' used; unable to detect undefined names

File: safe/definitions/post_processors/__init__.py [hunk 4]
```diff
@@ -1,9 +1,11 @@
...
 from post_processor_functions import *
+from post_processor_inputs import *
```
Comment (ismailsunni) [inline safe/definitions/post_processors/__init__.py]: __[flake8]__
* [F403] 'from post_processor_inputs import *' used; unable to detect undefined names

File: safe/definitions/post_processors/__init__.py [hunk 5]
```diff
@@ -1,9 +1,11 @@
...
+from post_processors import *
```
Comment (ismailsunni) [inline safe/definitions/post_processors/__init__.py]: __[flake8]__
* [F403] 'from post_processors import *' used; unable to detect undefined names

File: safe/definitions/post_processors/__init__.py [hunk 6]
```diff
@@ -42,6 +44,7 @@
      minimum_needs_post_processors +
      age_vulnerability_postprocessors +
      disabled_vulnerability_postprocessors +
-     gender_vulnerability_postprocessors) + [
-    post_processor_additional_rice
+     gender_vulnerability_postprocessors +
```
Comment (ismailsunni) [inline safe/definitions/post_processors/__init__.py]: __[flake8]__
* [F405] 'gender_vulnerability_postprocessors' may be undefined, or defined from star imports: minimum_needs_post_processors, population_post_processors, post_processor_functions, post_processor_inputs, post_processors

File: safe/definitions/post_processors/__init__.py [hunk 7]
```diff
@@ -42,6 +44,7 @@
...
+     productivity_post_processors) + [
+    post_processor_additional_rice,
```
Comment (ismailsunni) [inline safe/definitions/post_processors/__init__.py]: __[flake8]__
* [F405] 'post_processor_additional_rice' may be undefined, or defined from star imports: minimum_needs_post_processors, population_post_processors, post_processor_functions, post_processor_inputs, post_processors

**Comment by fredychandra** [comment]
Hi @ismailsunni 
Cool, I can try to test it and back to you for further discussion. 
Thanks

File: safe/definitions/fields.py [hunk 8]
```diff
@@ -2249,3 +2227,147 @@
...
+# Productivity field
+productivity_rate_field = {
+    'key': 'productivity_rate_field',
+    'name': tr('Productivity Rate'),
+    'field_name': 'productivity_rate',
+    'type': qvariant_numbers,
+    'length': default_field_length,
+    'precision': 0,
+    'absolute': True,
+    'help_text': tr('The rate of productivity of a crop in Kwt/Ha unit.'),
```
Comment (Gustry) [inline safe/definitions/fields.py]: What is `Kwt` ? Kilo-Watt Tons? Can we use the full name, because the acronym is not very well known.

Comment (ismailsunni) [inline safe/definitions/fields.py]: Ouch, sorry, I thought it is international term. It should be quintal (https://en.wikipedia.org/wiki/Quintal) I will add the concepts and update the tree.
Thanks

Comment (Gustry) [inline safe/definitions/fields.py]: Ok thanks. Let me know when you will add it. I will read and see if a not farmer like me can understand

File: safe/definitions/post_processors/__init__.py [hunk 9]
```diff
@@ -42,6 +47,7 @@
      minimum_needs_post_processors +
      age_vulnerability_postprocessors +
      disabled_vulnerability_postprocessors +
-     gender_vulnerability_postprocessors) + [
-    post_processor_additional_rice
+     gender_vulnerability_postprocessors +
```
Comment (ismailsunni) [inline safe/definitions/post_processors/__init__.py]: __[flake8]__
* [F405] 'gender_vulnerability_postprocessors' may be undefined, or defined from star imports: minimum_needs_post_processors, population_post_processors, post_processor_functions, post_processor_inputs, post_processors

File: safe/definitions/post_processors/__init__.py [hunk 10]
```diff
@@ -42,6 +47,7 @@
...
+     productivity_post_processors) + [
+    post_processor_additional_rice,
```
Comment (ismailsunni) [inline safe/definitions/post_processors/__init__.py]: __[flake8]__
* [F405] 'post_processor_additional_rice' may be undefined, or defined from star imports: minimum_needs_post_processors, population_post_processors, post_processor_functions, post_processor_inputs, post_processors

File: safe/definitions/fields.py (hunk 8 above)
Comment (ismailsunni) [inline safe/definitions/fields.py]: Hi @Gustry 
I have added the explanation in the last commit. I am not sure it's the best. May be we should ask @fredychandra or @samnawi to get a better explanation.

**Comment by samnawi** [comment]
LGTM

image
I just think we should add the unit (that InaSAFE used in the calculation) at the description on this step

**Comment by ismailsunni** [comment]
Thanks @samnawi 
I will merge this one after I added the unit and pass to @myarjunar to create the reporting.

File: safe/definitions/default_settings.py [hunk 11]
```diff
@@ -23,6 +24,8 @@
     'ISO19115_EMAIL': 'info@inasafe.org',
     'ISO19115_LICENSE': 'Free use with accreditation',
 
+    'currency': idr['key'],
```
Comment (Gustry) [inline safe/definitions/default_settings.py]: Are you trying to dominate the world with IDR? ;-)

Comment (ismailsunni) [inline safe/definitions/default_settings.py]: IDR has the biggest number compared the other :D 

Or should we ask the user what currency they want to use after installing InaSAFE? Or is it overkill :D

**Pull request merged**